sos/plugins/networking.py

```diff
@@ -201,13 +210,32 @@ def setup(self):
         cmd_prefix = "ip netns exec "
         if ip_netns['status'] == 0:
             out_ns = []
+            # Regex initialization outside of for loop
+            if self.get_option("namespaces"):
+                regex = '(?:%s)' % '|'.join(
+                        self.get_option("namespaces").split())
             for line in ip_netns['output'].splitlines():
                 # If there's no namespaces, no need to continue
                 if line.startswith("Object \"netns\" is unknown") \
                         or line.isspace() \
                         or line[:1].isspace():
                     continue
-                out_ns.append(line.partition(' ')[0])
+                # if namespaces defined, append only namespaces
+                # matching with regex
+                if self.get_option("namespaces"):
```

Comment:
A log message here would be helpful, E.G. `Limiting namespace iteration to first X namespaces found`.

That way if a support engineer is wondering why they're missing namespace details, it is easier to identify.

Reply:
@TurboTurtle Agree with log message, question is about log level, warn log level will show the message on screen, we can also use info level which will be only in sos_logs/sos.log. Which log level is more preferred in this case?

Reply:
Hmm, I'm leaning towards warn level but I'm not committed to it. It'll help with RHOSP preset runs where the user isn't explicitly setting the value themselves, but it'd still be good to alert them that we may not be capturing all available information.

Reply:
Warn level log was added for namespaces parameter.